Exiled.API/Features/Pickups/FirearmPickup.cs

```diff
@@ -45,6 +47,16 @@ internal FirearmPickup(ItemType type)
         /// </summary>
         public new BaseFirearm Base { get; }
 
+        /// <summary>
+        /// Gets the <see cref="Enums.FirearmType"/> of the firearm.
+        /// </summary>
+        public FirearmType FirearmType => Type.GetFirearmType();
```

Comment:
```suggestion
        public FirearmType FirearmType => firearmType ??= Type.GetFirearmType();
```

Add a `FirearmType firearmType` backing field as well, once it's set, it shouldn't change as the weapon instance will always be of the same type.